File: mobile/android/app/src/main/java/io/github/axuitomo/cfstgui/UpdatePackageCleanupReceiver.java
package io.github.axuitomo.cfstgui;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;
import java.io.File;

public class UpdatePackageCleanupReceiver extends BroadcastReceiver {
    private static final String TAG = "UpdatePackageCleanup";

    @Override
    public void onReceive(Context context, Intent intent) {
        if (context == null || intent == null || !Intent.ACTION_MY_PACKAGE_REPLACED.equals(intent.getAction())) {
            return;
        }
        try {
            CfstPlugin.cleanupAndroidUpdatePackages(new File(context.getFilesDir(), "updates"));
        } catch (Exception error) {
            Log.e(TAG, "Failed to clean Android update APK after package replacement.", error);
        }
    }
}

File: mobile/android/app/src/main/java/io/github/axuitomo/cfstgui/UpdatePackageCleanupReceiver.kt
package io.github.axuitomo.cfstgui

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class UpdatePackageCleanupReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent?) {
        if (context == null || intent?.action != Intent.ACTION_MY_PACKAGE_REPLACED) {
            return
        }
        try {
            AndroidUpdateInstaller.cleanupDownloadedPackages(context)
        } catch (error: Exception) {
            Log.e(TAG, "Failed to clean Android update APK after package replacement.", error)
        }
    }

    private companion object {
        private const val TAG = "UpdatePackageCleanup"
    }
}
